src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,18 +32,33 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={searchText}
+          onChange={(e) => {
+            dispatch(actions.setSearchText(e.target.value));
+            if (e.target.value !== '') {
+              setIsClear(true);
+            } else {
+              setIsClear(false);
+            }
+          }}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
-        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
+        <span className="icon is-right">
+          {isClear && (
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={() => {
+                dispatch(actions.setSearchText(''));
+                setIsClear(false);
+              }}
+              aria-label="Clear Search"
+            />
+          )}
         </span>
```

Comment:
I am not a fan of this setup, why is span wrapping a button? and why button does not have content?